### ISSUE TITLE
Can't seem to get /etc/hosts or AutoDNS to work

### ISSUE THREAD

**Issue by DamnedScholar** [cytopia/devilbox#695]
### ISSUE TYPE

 - Bug Report

### Checklist

* [x] `.env` file is attached - [env.txt](https://github.com/cytopia/devilbox/files/4446972/env.txt)
* [ ] `docker-compose.override.yml` is attached (if it exists)
* [ ] Custom configs from `cfg/` dir are attached (if customized)
* [x] `docker-compose logs` and `log/` output is added - [logs.txt](https://github.com/cytopia/devilbox/files/4446989/logs.txt)
* [x] I've looked through the docs: https://devilbox.readthedocs.io/en/latest/
* [x] I've looked through existing issues: https://github.com/cytopia/devilbox/issues
* [x] I've read troubleshooting: https://devilbox.readthedocs.io/en/latest/support/troubleshooting.html

### OS / ENVIRONMENT

1. Host operating system and version: Ubuntu 18.04 LTS
2. (Windows) Native Docker or Docker Toolbox: N/A
3. Docker version: 19.03.8, build afacb8b7f0
4. Docker Compose version: docker-compose version 1.25.4, build 8d51620a
5. (Linux) Is SELinux enabled?: No
6. What git commit hash are you on?: 8c931cd

### SUMMARY
I've been through all the documentation and seem to have followed every step to try both methods for telling Devilbox where to go for my project folder, but I haven't had any success.

### STEPS TO REPRODUCE
1. Installed Devilbox, made project folder, edited `/etc/hosts` as instructed.
```
in ~/github/devilbox/ on master± as damnedscholar
› cat /etc/hosts
127.0.0.1       c-street.loc

# The following lines are desirable for IPv6 capable hosts
::1     localhost ip6-localhost ip6-loopback
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
```
2. When that didn't work, I tried to bypass it and use `bind` (which is what I wanted to do at the end anyway). For this test, I edited `.env` to `HOST_PORT_BIND=53` and made sure that `127.0.0.1` was listed as a nameserver:
```
in ~/github/devilbox/ on master± as damnedscholar
› cat /etc/systemd/resolved.conf
#  This file is part of systemd.
#
#  systemd is free software; you can redistribute it and/or modify it
#  under the terms of the GNU Lesser General Public License as published by
#  the Free Software Foundation; either version 2.1 of the License, or
#  (at your option) any later version.
#
# Entries in this file show the compile time defaults.
# You can change settings by editing this file.
# Defaults can be restored by simply deleting this file.
#
# See resolved.conf(5) for details

[Resolve]
DNS=127.0.0.1 127.0.0.53
#FallbackDNS=
#Domains=
#LLMNR=no
#MulticastDNS=no
#DNSSEC=no
#Cache=yes
#DNSStubListener=yes
```
3. With Devilbox stopped, this is the `netstat` output:
```
in ~/github/devilbox/ on master± as damnedscholar
› netstat -an | grep -E 'LISTEN\s*$'
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN
tcp6       0      0 :::22                   :::*                    LISTEN
```

### EXPECTED BEHAVIOUR
The instructions don't seem complicated. If `/etc/hosts` is set correctly, there shouldn't be a red warning message for the virtual host.

### ACTUAL BEHAVIOUR
1. When `/etc/hosts` is set, Devilbox doesn't recognize it.
https://puu.sh/FuwGT/6127299ea5.png

2. When `HOST_PORT_BIND=53`, the server refuses to connect me to the Intranet at all.
```
in ~/github/devilbox/ on master± as damnedscholar
› nslookup c-street.loc
Server:         127.0.0.53
Address:        127.0.0.53#53

Non-authoritative answer:
Name:   c-street.loc
Address: 127.0.0.1
```
3. When Devilbox starts, `netstat` looks like this:
```
in ~/github/devilbox/ on master± as damnedscholar
› netstat -an | grep -E 'LISTEN\s*$'
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN
tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN
tcp6       0      0 :::22                   :::*                    LISTEN
tcp6       0      0 :::5432                 :::*                    LISTEN
tcp6       0      0 :::443                  :::*                    LISTEN
tcp6       0      0 :::1053                 :::*                    LISTEN
tcp6       0      0 :::11211                :::*                    LISTEN
tcp6       0      0 :::6379                 :::*                    LISTEN
tcp6       0      0 :::80                   :::*                    LISTEN
```

#### Start command

```
in ~/github/devilbox/ on master± as damnedscholar
› sudo docker-compose up httpd php bind pgsql redis memcd
```

#### File and user permissions (Linux & MacOS)

```
in ~/github/devilbox/ on master± as damnedscholar
› id
uid=1000(damnedscholar) gid=1000(damnedscholar) groups=1000(damnedscholar),27(sudo)
```

```
in ~/github/devilbox/ on master± as damnedscholar
› ls -la
total 264K
4.0K drwxrwxr-x 16 damnedscholar damnedscholar 4.0K 2020.04.07 21:02 ./
4.0K drwxrwxr-x  3 damnedscholar damnedscholar 4.0K 2020.03.29 22:02 ../
4.0K drwxrwxr-x  2 damnedscholar damnedscholar 4.0K 2020.03.29 22:02 autostart/
4.0K drwxr-xr-x  3 damnedscholar damnedscholar 4.0K 2020.03.29 22:12 backups/
4.0K drwxrwxr-x  2 damnedscholar damnedscholar 4.0K 2020.03.29 22:02 bash/
4.0K drwxrwxr-x  2 damnedscholar damnedscholar 4.0K 2020.03.29 22:12 ca/
4.0K drwxrwxr-x 57 damnedscholar damnedscholar 4.0K 2020.03.29 22:02 cfg/
4.0K drwxrwxr-x  2 damnedscholar damnedscholar 4.0K 2020.03.29 22:02 compose/
4.0K drwxrwxr-x  3 damnedscholar damnedscholar 4.0K 2020.03.29 22:02 data/
4.0K drwxrwxr-x  3 damnedscholar damnedscholar 4.0K 2020.03.29 22:02 .devilbox/
4.0K drwxrwxr-x 20 damnedscholar damnedscholar 4.0K 2020.03.29 22:02 docs/
4.0K drwxrwxr-x  8 damnedscholar damnedscholar 4.0K 2020.04.07 21:13 .git/
4.0K drwxrwxr-x  4 damnedscholar damnedscholar 4.0K 2020.03.29 22:02 .github/
4.0K drwxrwxr-x  6 damnedscholar damnedscholar 4.0K 2020.03.29 22:12 log/
4.0K drwxrwxr-x  9 damnedscholar damnedscholar 4.0K 2020.03.29 22:12 mod/
4.0K drwxrwxr-x  5 damnedscholar damnedscholar 4.0K 2020.03.29 22:02 .tests/
 12K -rw-rw-r--  1 damnedscholar damnedscholar 8.2K 2020.03.29 22:02 CHANGELOG.md
4.0K -rw-rw-r--  1 damnedscholar damnedscholar 2.8K 2020.03.29 22:02 CONTRIBUTING.md
4.0K -rw-rw-r--  1 damnedscholar damnedscholar  747 2020.03.29 22:02 docker-compose.override.yml-example
 16K -rw-rw-r--  1 damnedscholar damnedscholar  15K 2020.03.29 22:02 docker-compose.yml
4.0K -rw-rw-r--  1 damnedscholar damnedscholar 2.5K 2020.03.29 22:02 .editorconfig
 20K -rw-rw-r--  1 damnedscholar damnedscholar  19K 2020.04.07 20:59 .env
 20K -rw-rw-r--  1 damnedscholar damnedscholar  19K 2020.03.29 22:02 env-example
 20K -rw-rw-r--  1 damnedscholar damnedscholar  19K 2020.04.05 18:32 .env-first
4.0K -rw-rw-r--  1 damnedscholar damnedscholar 3.6K 2020.03.29 22:02 .gitignore
4.0K -rw-rw-r--  1 damnedscholar damnedscholar 1.1K 2020.03.29 22:02 LICENSE.md
 64K -rw-rw-r--  1 damnedscholar damnedscholar  64K 2020.03.29 22:02 README.md
4.0K -rw-rw-r--  1 damnedscholar damnedscholar   86 2020.03.29 22:02 shell.bat
4.0K -rwxrwxr-x  1 damnedscholar damnedscholar   59 2020.03.29 22:02 shell.sh*
 12K -rw-rw-r--  1 damnedscholar damnedscholar  12K 2020.03.29 22:02 .travis.yml
4.0K -rwxrwxr-x  1 damnedscholar damnedscholar 2.3K 2020.03.29 22:02 update-docker.sh*
4.0K -rw-rw-r--  1 damnedscholar damnedscholar 1.9K 2020.03.29 22:02 UPDATING.md
```

```
in ~/github/devilbox/ on master± as damnedscholar
› ls -la data/www/
total 12K
4.0K drwxr-xr-x 3 damnedscholar damnedscholar 4.0K 2020.03.29 23:28 ./
4.0K drwxrwxr-x 3 damnedscholar damnedscholar 4.0K 2020.03.29 22:02 ../
4.0K drwxrwxr-x 6 damnedscholar damnedscholar 4.0K 2020.04.05 17:03 c-street/
   0 -rw-rw-r-- 1 damnedscholar damnedscholar    0 2020.03.29 22:02 .keepme
```

**Comment by Natetronn** [comment]
Did you add 127.0.0.53? 

I don't think IP 127.0.0.53, on port 53, is going to work. Should just be 127.0.0.1 (which will then be on port 53.) Not sure you're editing the correct resolv file, either way. Not saying it can't work (I'm not 100% sure, as I'm on arch) but, I don't recall `/etc/systemd/resolved.conf` being mentioned.

For the .evn file, should be set to `127.0.0.1:` (note the trailing colon.)
https://devilbox.readthedocs.io/en/latest/configuration-files/env-file.html#env-local-listen-addr

You appear to have the port (53) set correctly in the .env but, this isn't correct (note my initial paragraph):

`tcp        0      0 127.0.0.53:53           0.0.0.0:*               LISTEN`

It should look like this, per the docs:

`tcp        0      0 127.0.0.1:53            0.0.0.0:*               LISTEN`

How to edit per project dns:
https://devilbox.readthedocs.io/en/latest/howto/dns/add-project-dns-entry-on-linux.html

Auto DNS and its prerequisites:
https://devilbox.readthedocs.io/en/latest/intermediate/setup-auto-dns.html#prerequisites
https://devilbox.readthedocs.io/en/latest/howto/dns/add-custom-dns-server-on-linux.html

**Comment by ianmjones** [comment]
I had the same problem getting AutoDNS to work on an Ubuntu 18.04 LTS based distribution (elementary 5.1.5).

The problem stems from Ubuntu 18.04+ using systemd-resolved for DNS by default, which is where that `127.0.0.53:53` address comes from. That stops devilbox from being able to bind to `0.0.0.0:53`.

To combat this, I just needed to do two things...

1. Ensure devilbox's `.env` was **only** listening on `127.0.0.1` by setting `LOCAL_LISTEN_ADDR=127.0.0.1:`.

2. Update my network connection to ensure `127.0.0.1` was used as the first name server before my router's. To do that, I switched my machines IPv4 connection to be "Automatic (DHCP) addresses only" rather than the usual "Automatic (DHCP)", and then entered `127.0.0.1, 192.168.1.1` as the additional DNS servers (`192.168.1.1` is my router's IP address).

![Screenshot from 2020-07-01 11-08-50@2x](https://user-images.githubusercontent.com/4710/86232165-7cfc9e00-bb8b-11ea-986b-4afefc929fd5.png)

After saving that, then disconnecting and reconnecting that network interface (just via the GUI), all worked fine with AutoDNS and there was no need to manually add entries into `/etc/hosts`.
